lib/rules/boolean-prop-naming.js

```diff
@@ -172,9 +172,10 @@ module.exports = {
      * @param {Function} addInvalidProp callback to run for each error
      */
     function runCheck(proptypes, addInvalidProp) {
-      proptypes = proptypes || [];
+      let newProptypes = proptypes;
+      newProptypes = newProptypes || [];
```

Comment:
```suggestion
```

Reply:
I'm a bit unclear as to why I should remove this?

Reply:
The argument (`proptypes`) is only actually needed once - in the forEach - and the `|| []` logic is basically the same conceptually as "only try to iterate the array if the argument is truthy".

So, one alternative way to write it is:
```js
(argument || []).forEach…
```
and another would be:
```js
if (argument) {
  argument.forEach…
}
```

and in neither case do we need to a) reassign to an argument, or b) create a variable at all (since it's only used once). Either case reads simpler and cleaner to me than what's in master.

Reply:
Done.